rfcs/20250618-pr-rules/README.md

```diff
@@ -0,0 +1,207 @@
+# oneDAL Pull Request Rules
+
+## Introduction
+
+Generating, reviewing and merging code in oneDAL are key pillars in providing
+a high quality and performant product.  The oneDAL codebase is both large and
+intricate requiring domain knowledge and precision.  In order to guarantee these
+qualities in oneDAL for all contributors, a set of rules should be added.
+
+
+## Proposal
+
+This enumerates in detail the aspects and proceedures necessary to getting a
+Pull Request (PR) merged into oneDAL.  This (as listed in the preamble below)
+will assist new developers as well as standardize for all developers in
+completing the necessary steps. The goal would be to add this to oneDAL's
+public documentation and refer to it in the template (as a link) to all future
+Pull Requests to oneDAL.
+
+# Text to be added:
+
+## Preamble:
+
+This is the rules for merging PRs into the oneDAL code base on Github. This
+gives the minimum requirements for PR submitters and reviewers. The goal is
+to enumerate the steps for better efficiency in oneDAL development.
+
+## Section 1: Opening a PR, Ready for Review
+
+1. PR Template and Task Checklist
+   * All PRs must use the provided template, including the task checklist.
```

Comment:
Let's mention checklist scope definition - currently it's assume people will be removing not relevant checkboxes. 
Might be instead limit them to minimal set, providing option for selecting  is some of those are not applicable? 

https://github.com/conda-forge/intel_repack-feedstock/pull/98